jcl/src/java.base/share/classes/java/lang/invoke/VarHandle.java

```diff
@@ -287,7 +358,147 @@ public static AccessMode valueFromMethodName(String methodName) {
 		this.handleTable = handleTable;
 		this.modifiers = modifiers;
 	}
-	
+
+/*[IF Java14]*/
+	/**
+	 * Constructs a generic VarHandle instance.
+	 *
+	 * @param varForm an instance of VarForm.
+	 */
+	VarHandle(VarForm varForm) {
+		AccessMode accessModes[] = AccessMode.values();
+		int numAccessModes = accessModes.length;
+
+		MethodType[] operationMTs = new MethodType[numAccessModes];
+		MethodType[] operationMTsExact = null;
+
+		MethodType getter = null;
+		MethodType setter = null;
+		
+		Class<?> operationsClass = null;
+		
+		/* The first argument in AccessType.GET MethodType is the receiver class. */
+		Class<?> receiverActual = accessModeTypeUncached(AccessMode.GET).parameterType(0);
+		Class<?> receiverVarForm = varForm.methodType_table[AccessType.GET.ordinal()].parameterType(0);
+		
+		/* Specify the exact operation method types if the actual receiver doesn't match the
+		 * receiver derived from VarForm.
+		 */
+		if (receiverActual != receiverVarForm) {
+			operationMTsExact = new MethodType[numAccessModes];
+		}
+		
+		for (int i = 0; i < numAccessModes; i++) {
+			MemberName memberName = varForm.memberName_table[i];
+			if (memberName != null) {
+				Method method = memberName.method;
+				if (method != null) {
+					operationMTs[i] = MethodType.methodType(method.getReturnType(), method.getParameterTypes());
```

Comment:
No, this data is not already available in `VarForm`. `VarForm` has a list of `MemberName`s (one for each access mode) and a list of `MethodTypes` (one for each access type).

Reply:
In the future, when we fully implement `MemberName` we should probably look at this again to see if it's still necessary or can be done more efficiently

Reply:
@fengxue-IS plans to use OpenJDK `MemberName` and `VarHandle` via https://github.com/eclipse/openj9/pull/7680, which is part of https://github.com/eclipse/openj9/issues/7352. Minimal effort was put from an optimization side since these changes will be removed in the future.